src/sql/src/func.rs

```diff
@@ -1664,7 +1664,21 @@ lazy_static! {
 
             // Aggregates.
             "array_agg" => Aggregate {
-                params!(Any) => Operation::unary(|_ecx, _e| unsupported!("array_agg")), 4053;
+                params!(NonVecAny) => Operation::unary(|ecx, e| {
+                    // TODO(#7572): remove this
+                    let e = match ecx.scalar_type(&e) {
+                        ScalarType::Char { length } => e.call_unary(UnaryFunc::PadChar { length }),
+                        _ => e,
+                    };
```

Comment:
Is this legit? I thought we couldn't have char arrays atm.

Reply:
I think the simplest thing to do here is just error if encountering `char` datums.

This will not have the right semantics if `length.is_none()`, which can happen if you're dealing with `char` datums that have been filtered through a `char[]`--so I imagine there's some way of calling this function twice to get it to break the char semantics.

fwiw the right place for this prohibition on `char[]` to error is probably `ArrayCreate`, but I currently have the error occurring during planning; will send up a patch for that and if Matt rebases on top of that, this code could be left-as is because the function call will bail.

Reply:
Oh hmm--making this an eval error on `VariadicFunc::ArrayCreate` when encountering `ScalarType::Char` feels bad because it will plan the statement knowing full well it will always error, and then provide an unfortunate surprise e.g if used in a view.

You feel comfortable just erroring in planning function call, @mjibson? Is a temporary bandaid.

Reply:
That’s what I was going to recommend too!

Reply:
I cargo culted this line. Happy to just error on it instead.